### ISSUE TITLE
combination of ifconfig:: and include:: doesn't work after subsection definition

### ISSUE THREAD

**Issue by tlingner** [sphinx-doc/sphinx#6213]
Hi, when I combine ifconfig:: and include:: (conditional inclusion of subdocuments) it works most of the time as expected, but after definition of the first subsection it doesn't.

I defined a config variable in conf.py:
```
def setup(app):
    app.add_config_value('testSwitch', 0, 'dev')
testSwitch = 1
```

The RST document looks like this:
```
Module A
========

.. ifconfig:: testSwitch == 0

   .. include:: inclusion.rst

Section 1
---------

Hi, I'm always there

.. ifconfig:: testSwitch == 1

   .. include:: inclusionMonly.rst

Section bla
-----------

Hi, I'm always there
```

I expect the content of the second file (inclusionMonly.rst) to be included in the HTML, but it's not. Log output is normal without any errors. When I change to section it is shown as expected, when I remove subsection it is shown as expected. Normal text (without include) also works as expected.

Project code:
[ModularTest.zip](https://github.com/sphinx-doc/sphinx/files/3013805/ModularTest.zip)

**Environment info**
- OS: Ubuntu 18.04
- Python version: 3.6.1
- Sphinx version: 1.8.2
- Sphinx extensions:  ifconfig

Thanks,
Thomas

**Comment by tk0miya** [comment]
ifconfig does not support sectioning. So it would be better to emit warnings for such case.

**Comment by tk0miya** [comment]
Oh, sorry. My comment above was wrong. ifconfig module supports headings. So something goes wrong in this case. I'll take a look later.

**Comment by tk0miya** [comment]
Fixed by #6254. But it fixes only crashes. Sphinx still does not support modifying sectioning via extensions. So that is not still unsupported officially.
Thanks,
